internal/impl/awk/processor.go

```diff
@@ -590,6 +604,14 @@ var awkFunctionsMap = map[string]any{
 	"print_log": func(value, level string) {
 		// Do nothing, this is a placeholder for compilation.
 	},
+	"base64_encode": func(data string) string {
+		// Do nothing, this is a placeholder for compilation.
```

Comment:
We only need placeholders for functions that interact directly with messages, since this is processing the argument directly we can just add the implementations here and ignore the customFuncs lower down.

Reply:
Spot on! I was trying to figure out how these needs to work and then forgot to check if I still need the boilerplate. Thanks, fixed! 😽